### PR TITLE
LPD-23987 rebase

### PR DESCRIPTION
**Pull request by daiazius**
Previous PR: https://github.com/liferay-appsec/liferay-portal/pull/1851

Continuing @manuelecastro work as she isn't available. I've only rebased the previous PR making the changes @stian-sigvartsen asked.

### CODE REVIEW

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -1450,7 +1450,7 @@ allowed-classes=Allowed Classes
 allowed-credentials-per-user=Allowed Credentials Per User
 allowed-credentials-per-user-description=Set the number of allowed credentials per user.
 allowed-domains=Allowed Domains
-allowed-domains-help=Set the domains which the portal is allowed to redirect to. Remove all entries to allow any domain. Specifying a domain with a leading "*." allows redirects to subdomains. PORTAL_DOMAINS allows redirects to all portal's virtual hosts.
+allowed-domains-help=Set the domains which the portal is allowed to redirect to. Remove all entries to allow any domain. Specifying a domain with a leading "*." allows redirects to subdomains. PORTAL_DOMAIN will be replaced with the current domain of the portal or site.
```
Comment (daiazius) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:1453]: Changed the language key to Stian's suggestion

File: portal-impl/test/unit/com/liferay/portal/util/PortalImplEscapeRedirectTest.java [hunk 2]
```diff
@@ -129,6 +130,20 @@ public void testEscapeRedirectWithDomains() throws Exception {
...
+			_portalImpl.escapeRedirect(
+				"https://" + _HOSTNAME_PORTAL_DOMAIN + ":8080"));
+
+		// Allow Virtual Hosts
+
+		Assert.assertEquals(
+			"https://" + _HOSTNAME_VIRTUAL_HOST + ":8080",
+			_portalImpl.escapeRedirect(
+				"https://" + _HOSTNAME_VIRTUAL_HOST + ":8080"));
+
```
Comment (daiazius) [inline portal-impl/test/unit/com/liferay/portal/util/PortalImplEscapeRedirectTest.java:146]: Grouped the new tests here, separated into two sections starting with the same word, just like Brian asked

**Comment by daiazius** [comment]
ci:test:sf

**Review by stian-sigvartsen — APPROVED** [review]
Perfect! Thanks @daiazius

**Comment by stian-sigvartsen** [comment]
ci:forward

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:forward:force

**Comment by stian-sigvartsen** [comment]
ci:forward:force

**Comment by stian-sigvartsen** [comment]
ci:forward:force

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by manuelecastro** [comment]
Hi @stian-sigvartsen @daiazius ! 

It looks like stable suite is failing due to the code changes. `PortalSmoke#Smoke` is throwing `LIFERAY_ERROR: Redirect URL http://test-1-37-70.lax.liferay.com:8080/... is not allowed`. see: [Jenkins Results](https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1162272/testReport/junit/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_PortalSmoke_Smoke_/)

I'm finding this strange because it did not fail on this previous PR: https://github.com/liferay-appsec/liferay-portal/pull/1842. Maybe there was a change to the PortalSmoke test...

**Comment by daiazius** [comment]
I was discussing the error with @manuelecastro and we found out that the PR was missing a commit, I added it and now everything should work

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:forward

**Comment by stian-sigvartsen** [comment]
Great job @daiazius @manuelecastro !

**Pull request closed without merging**